### PR TITLE
Mostly finished ShoppingList + Code fixes + Animation

### LINKED ISSUE #17: US 04.01.01 - As a meal planner, I want to view a shopping list of ingredients that I need to buy, based on my ingredient storage and meal plan so that I can make sure I have all ingredients necessary to complete my meal plan

### PR DESCRIPTION
**Pull request by kwiskel**
closes #17 #20 #21

### CODE REVIEW

**Comment by kwiskel** [comment]
Also added: Comment, javadocs, fixed code, bug fixes for ShoppingLis

File: android/301Project/app/src/main/java/com/example/a301project/ShoppingListAdapter.java [hunk 1]
```diff
@@ -46,24 +57,43 @@ public ShoppingListAdapter(Context context, ArrayList<ShoppingItem> shoppingItem
...
 
         // sets the text
         shoppingItemName.setText(s.getName());
-        amountName.setText(s.getAmount().toString());
+        amountName.setText("Need: " + s.getAmount().toString());
         unitName.setText(s.getUnit());
         categoryName.setText(s.getCategory());
 
+        // if the test is null -> don't display the text
+        if (s.getUnit() == "null") {
```
Comment (yismailuofa) [inline android/301Project/app/src/main/java/com/example/a301project/ShoppingListAdapter.java:95]: Is this supposed to be `null` or `"null" `

File: android/301Project/app/src/main/java/com/example/a301project/ShoppingListFragment.java [hunk 2]
```diff
@@ -140,4 +163,43 @@ private void sortDataBySpinner() {
...
+        // open the add/edit fragment but the "SHOPPING" tag
+        AddEditIngredientFragment.newInstance(selected,false, ShoppingListFragment.this).show(getChildFragmentManager(),"SHOPPING");
+    }
+
+    @Override
+    public void f(ArrayList<ShoppingItem> r) {
+        listCount++;
+        if (listCount == 1) {
+
+        }
```
Comment (yismailuofa) [inline android/301Project/app/src/main/java/com/example/a301project/ShoppingListFragment.java]: Not sure if intended but Empty if statement

**Review by yismailuofa — APPROVED** [review]

File: android/301Project/app/src/main/java/com/example/a301project/ShoppingListAdapter.java (hunk 1 above)
Comment (yismailuofa) [reply android/301Project/app/src/main/java/com/example/a301project/ShoppingListAdapter.java:95]: Intended to be "null"

**Pull request merged**